uint8[] data        # Serialized encrypted data as byte array
uint32 data_type    # 0: x coordinate, 1: y coordinate, 2: sum(x+y), 3: prod(x*y)